lib/parsec/util.py

```diff
@@ -129,32 +133,58 @@ def m_override( target, sparse ):
     """Override items in a target pdict. Target keys must already exist
     unless there is a "__MANY__" placeholder in the right position."""
     if not sparse:
-        target = OrderedDict()
+        target = OrderedDictWithDefaults()
         return
-    for key,val in sparse.items():
-        if isinstance( val, dict ):
-            if key not in target:
-                if '__MANY__' in target:
-                    target[key] = OrderedDict()
-                    replicate( target[key], target['__MANY__'] )
+    stack = [(sparse, target, [], OrderedDictWithDefaults())]
```

Comment:
I put it in a stack because the recursion was making me confused, and I needed the defaults setting at the very end - but it's actually not very different from the original function.